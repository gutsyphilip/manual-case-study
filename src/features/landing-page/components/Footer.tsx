import React from "react";
import { Link } from "react-router-dom";

import {
  ManualLogo,
  FacebookIcon,
  TwitterIcon,
  LinkedInIcon,
} from "../../../assets/svgs";

import styles from "./landing-page.module.scss";

interface Props {}

const Footer = (props: Props) => {
  return (
    <footer className={styles.footer}>
      <section className={styles.footer_main}>
        <ManualLogo className={styles.footer_main_logo} />
        <nav className={styles.footer_main_nav}>
          <div className={styles.footer_main_nav_group}>
            <header className={styles.footer_main_nav_group_header}>
              <h5 className={styles.footer_main_nav_group_header_title}>
                PRODUCT
              </h5>
            </header>
            <ul className={styles.footer_main_nav_group_list}>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Popular</Link>
              </li>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Trending</Link>
              </li>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Guided</Link>
              </li>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Products</Link>
              </li>
            </ul>
          </div>
          <div className={styles.footer_main_nav_group}>
            <header className={styles.footer_main_nav_group_header}>
              <h5 className={styles.footer_main_nav_group_header_title}>
                Company
              </h5>
            </header>
            <ul className={styles.footer_main_nav_group_list}>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Press Releases</Link>
              </li>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Mission</Link>
              </li>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Strategy</Link>
              </li>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">About</Link>
              </li>
            </ul>
          </div>
          <div className={styles.footer_main_nav_group}>
            <header className={styles.footer_main_nav_group_header}>
              <h5 className={styles.footer_main_nav_group_header_title}>
                INFO
              </h5>
            </header>
            <ul className={styles.footer_main_nav_group_list}>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Support</Link>
              </li>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Customer Service</Link>
              </li>
              <li className={styles.footer_main_nav_group_list_item}>
                <Link to="">Get Started Guide</Link>
              </li>
            </ul>
          </div>
          <div className={styles.footer_main_nav_group}>
            <header className={styles.footer_main_nav_group_header}>
              <h5 className={styles.footer_main_nav_group_header_title}>
                Follow us
              </h5>
            </header>
            <ul className={styles.footer_main_nav_group_icons}>
              <li className={styles.footer_main_nav_group_icons_icon}>
                <Link to="">
                  <FacebookIcon />
                </Link>
              </li>
              <li className={styles.footer_main_nav_group_icons_icon}>
                <Link to="">
                  <TwitterIcon />
                </Link>
              </li>
              <li className={styles.footer_main_nav_group_icons_icon}>
                <Link to="">
                  <LinkedInIcon />
                </Link>
              </li>
            </ul>
          </div>
        </nav>
      </section>
      <div className={styles.footer_sub}>
        <hr />
        <p className={styles.footer_sub_credits}>
          &copy; 2019 Manual. All rights reserved.
        </p>
      </div>
    </footer>
  );
};

export default Footer;
