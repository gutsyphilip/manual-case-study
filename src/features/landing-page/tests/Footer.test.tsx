import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import Footer from "../components/Footer";

const FooterProps = {};
test("renders landing page footer", () => {
  const tree = render(
    <BrowserRouter>
      <Footer {...FooterProps} />
    </BrowserRouter>
  );
  expect(tree.asFragment()).toMatchSnapshot();
});
