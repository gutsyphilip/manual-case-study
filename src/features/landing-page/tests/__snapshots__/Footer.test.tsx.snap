// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders landing page footer 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <section
      class="footer_main"
    >
      <svg
        class="footer_main_logo"
      >
        manual-logo.svg
      </svg>
      <nav
        class="footer_main_nav"
      >
        <div
          class="footer_main_nav_group"
        >
          <header
            class="footer_main_nav_group_header"
          >
            <h5
              class="footer_main_nav_group_header_title"
            >
              PRODUCT
            </h5>
          </header>
          <ul
            class="footer_main_nav_group_list"
          >
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Popular
              </a>
            </li>
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Trending
              </a>
            </li>
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Guided
              </a>
            </li>
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Products
              </a>
            </li>
          </ul>
        </div>
        <div
          class="footer_main_nav_group"
        >
          <header
            class="footer_main_nav_group_header"
          >
            <h5
              class="footer_main_nav_group_header_title"
            >
              Company
            </h5>
          </header>
          <ul
            class="footer_main_nav_group_list"
          >
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Press Releases
              </a>
            </li>
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Mission
              </a>
            </li>
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Strategy
              </a>
            </li>
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                About
              </a>
            </li>
          </ul>
        </div>
        <div
          class="footer_main_nav_group"
        >
          <header
            class="footer_main_nav_group_header"
          >
            <h5
              class="footer_main_nav_group_header_title"
            >
              INFO
            </h5>
          </header>
          <ul
            class="footer_main_nav_group_list"
          >
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Support
              </a>
            </li>
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Customer Service
              </a>
            </li>
            <li
              class="footer_main_nav_group_list_item"
            >
              <a
                href="/"
              >
                Get Started Guide
              </a>
            </li>
          </ul>
        </div>
        <div
          class="footer_main_nav_group"
        >
          <header
            class="footer_main_nav_group_header"
          >
            <h5
              class="footer_main_nav_group_header_title"
            >
              Follow us
            </h5>
          </header>
          <ul
            class="footer_main_nav_group_icons"
          >
            <li
              class="footer_main_nav_group_icons_icon"
            >
              <a
                href="/"
              >
                <svg>
                  facebook.svg
                </svg>
              </a>
            </li>
            <li
              class="footer_main_nav_group_icons_icon"
            >
              <a
                href="/"
              >
                <svg>
                  twitter.svg
                </svg>
              </a>
            </li>
            <li
              class="footer_main_nav_group_icons_icon"
            >
              <a
                href="/"
              >
                <svg>
                  linkedin.svg
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </section>
    <div
      class="footer_sub"
    >
      <hr />
      <p
        class="footer_sub_credits"
      >
        Â© 2019 Manual. All rights reserved.
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
